pkg/secrets-store/provider_client_test.go

```diff
@@ -511,3 +511,38 @@ func TestPluginClientBuilder_HealthCheck(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestIsMaxRecvMsgSizeError(t *testing.T) {
+	cases := []struct {
+		name string
+		// inputs
+		err error
+		// expectations
+		expectedBool bool
+	}{
+		{
+			name:         "not resource exhausted error",
+			err:          errors.New("failed to mount"),
+			expectedBool: false,
+		},
+		{
+			name:         "resource exhausted error because of quota propagation",
+			err:          status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max length allowed on current machine"),
+			expectedBool: false,
+		},
+		{
+			name:         "resource exhausted error because message larger than max length",
+			err:          status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max"),
+			expectedBool: true,
+		},
+	}
+
+	for _, test := range cases {
+		t.Run(test.name, func(t *testing.T) {
+			actualBool := isMaxRecvMsgSizeError(test.err)
+			if actualBool != test.expectedBool {
+				t.Fatalf("expected: %v, got: %v", test.expectedBool, actualBool)
```

Comment:
`Errorf` instead of `Fatalf`: https://github.com/golang/go/wiki/TestComments#keep-going

"isMaxRecvMsgSizeError() = %v, want %v": https://github.com/golang/go/wiki/TestComments#identify-the-function